### PR TITLE
211 inputs

### PR DESCRIPTION
**Pull request by dupreesi**
#### Issues this PR relates to
#211 
- adds inputs
- updates storybook preview.js to render and update states
---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

- [ ] npm i 
- [ ] cd client && npm run storybook
- [ ] check all inputs: Inputfields, datepicker, counter, dropdowns, checkboxes
- [ ] check designs AND correct state update

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -10,10 +10,12 @@
     "axios": "^0.19.2",
     "prop-types": "^15.7.2",
     "react": "^16.13.1",
+    "react-dates": "^21.8.0",
     "react-dom": "^16.13.1",
     "react-redux": "^7.2.0",
     "react-router-dom": "^5.1.2",
     "react-scripts": "3.4.1",
+    "react-select": "^3.1.0",
```
Comment (RamyAlshurafa) [inline client/package.json:18]: why you didn't use material UI components? they don't match our designs?

File: client/src/components/Inputs/index.stories.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
+import React from "react";
```
Comment (RamyAlshurafa) [inline client/src/components/Inputs/index.stories.js:1]: file name `Inputs.stories.js` make more sense

File: client/src/components/Inputs/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import InputField from "./InputField/InputField";
```
Comment (RamyAlshurafa) [inline client/src/components/Inputs/index.js:1]: if you  named your files like this `InputField/index.js`
then you will be able to `import InputField from "./InputField";`

File: client/package.json (hunk 1 above)
Comment (RamyAlshurafa) [reply client/package.json:18]: same for `react-dates`

Comment (dupreesi) [reply client/package.json:18]: @RamyAlshurafa no none of the material UI components match our designs and need a lot of work to get adjusted to fit those. Its more about functionality for multiple dropdowns with checkboxes and date ranges. 

both react-select and air bnb react-dates are pretty light weight and very well maintained libraries.

File: client/src/components/Inputs/Datepicker/react_dates_overrides.css [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/*
```
Comment (fadeomar) [inline client/src/components/Inputs/Datepicker/react_dates_overrides.css:1]: un used file if you can remove it plz

File: client/src/components/Inputs/InputField/index.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+              onClick={handleClickShowPassword}
+              onMouseDown={handleMouseDownPassword}
+              aria-label="toggle password visibility"
+            >
+              <Icon width="40px" icon="eye" />
+            </IconButton>
+          </InputAdornment>
+        ),
+      }}
+    />
```
Comment (fadeomar) [inline client/src/components/Inputs/InputField/index.js:107]: i should be able to pass additional props like name, id, onfocus ..
```suggestion
    />
```
```suggestion
{...props}
    />
```

**Comment by fadeomar** [comment]
i should be able to pass additional props like name, id, onfocus ..
```suggestion
{...props}
    />
```

**Pull request merged**